code_annotations/report_templates/annotation_list.tpl

```diff
@@ -0,0 +1,29 @@
+{% extends "base.tpl" %}
+{% block content %}
+Complete Annotation List
+------------------------
+Annotations found in {{ report|length }} files.
+
+{% for filename in report %}
+{{ filename }} has {{ report[filename]|length }} annotations.
+
+    {% for annotation in report[filename] %}
+        {% if annotation.report_group_id %}
+            {% if loop.changed(annotation.report_group_id) %}
+                {% include 'annotation_group.tpl' %}
+
+            {% endif %}
+    * {% include 'annotation.tpl' %}
+
+        {% else %}
+            {% if loop.changed(annotation.report_group_id) %}
+
+            {% endif %}
```

Comment:
Can you clarify what the above if-block does exactly?  Is the extra newline necessary to terminate the group because un-indenting isn't enough to cue the termination?

Reply:
Yes, unfortunately handling rst whitespace in Jinja2 templates is super finicky and trying to comment them only makes it worse.